test/hotspot/jtreg/compiler/arguments/TestMethodArguments.java

```diff
@@ -0,0 +1,160 @@
+                Template.let("boxedType", t.boxedTypeName()),
+                Template.let("arguments", arguments),
+                Template.let("parameters", parameters),
+                Template.let("sum", sum),
+                Template.let("inputSize", INPUT_SIZE),
+                Template.let("numberOfArguments", numberOfArguments),
+                """
+                        static #boxedType[][] $inputs = generateInput(generator#boxedType, new #boxedType[#inputSize][#numberOfArguments]);
+                        static int $nextInput = 0;
+                        static #type[] $golden = $init();
+
+                        public static #type[] $init() {
+                            #type[] golden = new #type[$inputs.length];
+                            for (int i = 0; i < golden.length; ++i) {
+                                #boxedType[] input = $inputs[i];
+                                golden[i] = $test(#arguments);
+                            }
+                            return golden;
+                        }
+
+                        @Setup
+                        public static Object[] $setup() {
+                            #boxedType[] input = $inputs[$nextInput];
+                            return new Object[]{#arguments};
+                        }
+
+                        @Test
+                        @Arguments(setup = "$setup")
+                        public static #type $test(#parameters) {
+                            return #sum;
+                        }
+
+                        @Check(test = "$test")
+                        public static void $check(#type res) {
+                            if (res != $golden[$nextInput]) {
+                                throw new RuntimeException("wrong result " + res + "!=" + $golden[$nextInput]);
+                            }
+                            $nextInput = ($nextInput + 1) % $inputs.length;
+                        }
+                        """));
+    }
+
+    public static String generate(CompileFramework comp) {
+        List<Object> tests = new LinkedList<>();
+        for (int i = MIN; i <= MAX; ++i) {
+            tests.add(generateTest(CodeGenerationDataNameType.ints(), i).asToken());
+            tests.add(generateTest(CodeGenerationDataNameType.floats(), i).asToken());
+            // Longs and doubles take up double as much space in the parameter list as other
+            // primitive types (e.g., int). We therefore have to divide by two to fill up
+            // the same amount of space as for ints and floats.
+            tests.add(generateTest(CodeGenerationDataNameType.longs(), i / 2).asToken());
+            tests.add(generateTest(CodeGenerationDataNameType.doubles(), i / 2).asToken());
+        }
+        return Template.make(() -> Template.body(
+                Template.let("classpath", comp.getEscapedClassPathOfCompiledClasses()),
+                """
+                        import java.util.Arrays;
```

Comment:
Personally, I would not indent this deeply. I know that the generated code will not have proper indentation, but that's no so bad. Readability of the Templates is more important I think. Subjective though.

Reply:
OK, what I meant is that I did not understand how exactly you wanted me to make the test more readable. But, I had a look at the template framework example and will update to use the same style (align the `"""`)!